src/server/game/Entities/Item/Item.cpp

```diff
@@ -962,6 +962,17 @@ bool Item::GemsFitSockets() const
     return true;
 }
 
+bool Item::HasAnySocket() const
+{
+    return
+        // Built-in sockets.
+        this->GetTemplate()->Socket[0].Color
+
+        // Socket from upgrade.
+        // There can only be one socket added, and it's always in slot `PRISMATIC_ENCHANTMENT_SLOT`.
+        || this->GetEnchantmentId(EnchantmentSlot(PRISMATIC_ENCHANTMENT_SLOT));
```

Comment:
Also this I'd rather see the code altogether and the comments above instead of dragging the comments and code onto 6 lines.
Something like this
```cpp
// There can only be one socket added, and it's always in slot `PRISMATIC_ENCHANTMENT_SLOT`.
// Built-in sockets                      Socket from upgrade          
this->GetTemplate()->Socket[0].Color || this->GetEnchantmentId(EnchantmentSlot(PRISMATIC_ENCHANTMENT_SLOT));